utils/utils.py

```diff
@@ -173,42 +232,108 @@ def dataset_loading_and_splitting(
                 dataset=dataset_combined,
                 batch_size=config["batch_size"],
                 perc_train=config["perc_train"],
+                distributed_data_parallelism=distributed_data_parallelism,
             )
 
 
-def split_dataset(dataset: [], batch_size: int, perc_train: float):
+def split_dataset(
+    dataset: [],
+    batch_size: int,
+    perc_train: float,
+    distributed_data_parallelism: bool = False,
+):
     perc_val = (1 - perc_train) / 2
     data_size = len(dataset)
-    train_loader = DataLoader(
-        dataset[: int(data_size * perc_train)], batch_size=batch_size, shuffle=True
-    )
-    val_loader = DataLoader(
-        dataset[int(data_size * perc_train) : int(data_size * (perc_train + perc_val))],
-        batch_size=batch_size,
-        shuffle=True,
-    )
-    test_loader = DataLoader(
-        dataset[int(data_size * (perc_train + perc_val)) :],
-        batch_size=batch_size,
-        shuffle=True,
-    )
+
+    trainset = dataset[: int(data_size * perc_train)]
+    valset = dataset[
+        int(data_size * perc_train) : int(data_size * (perc_train + perc_val))
+    ]
+    testset = dataset[int(data_size * (perc_train + perc_val)) :]
+
+    if distributed_data_parallelism:
+
+        train_sampler = torch.utils.data.distributed.DistributedSampler(trainset)
+        val_sampler = torch.utils.data.distributed.DistributedSampler(valset)
+        test_sampler = torch.utils.data.distributed.DistributedSampler(testset)
+
+        train_loader = DataLoader(
+            trainset, batch_size=batch_size, shuffle=False, sampler=train_sampler
+        )
+        val_loader = DataLoader(
+            trainset, batch_size=batch_size, shuffle=False, sampler=val_sampler
+        )
+        test_loader = DataLoader(
+            trainset, batch_size=batch_size, shuffle=False, sampler=test_sampler
+        )
+
+    else:
+
+        train_loader = DataLoader(trainset, batch_size=batch_size, shuffle=True)
+        val_loader = DataLoader(
+            valset,
+            batch_size=batch_size,
+            shuffle=True,
+        )
+        test_loader = DataLoader(
+            testset,
+            batch_size=batch_size,
+            shuffle=True,
+        )
 
     return train_loader, val_loader, test_loader
 
 
 def combine_and_split_datasets(
-    dataset1: [], dataset2: [], batch_size: int, perc_train: float
+    dataset1: [],
```

Comment:
May as well rename these `trainset` and `testset` (unless you're worried about backwards compatibility)

Reply:
actually there are 3 partitioning performed on 2 datasets

dataset1 is split into training and validation. dataset 2 becomes the testset. 
So renaming dataset1 as trainingset may be confusing